### PR TITLE
[E2E] Fix auth file playwright and small refactor

### PR DESCRIPTION
**Pull request by yellowee**
## What type of PR is this?
- [ ] 💅 Refactor
- [ ] 🌟 Feature
- [ ] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes #

## Have you written tests?
- [ ] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

### CODE REVIEW

**Comment by yellowee** [comment]
TC 2111 failure is a known issue and will be fixed in https://linear.app/saleor/issue/QAG-146/fix-failing-tests-in-runs-on-pr-environments

File: playwright/api/basics.ts [hunk 1]
```diff
@@ -51,6 +51,8 @@ export class BasicApiService {
     });
     const loginResponseJson = await loginResponse.json();
 
-    return loginResponseJson as ApiResponse<TokenCreateResponse>;
+    expect(loginResponseJson.data.tokenCreate.errors).toEqual([]);
```
Comment (szczecha) [inline playwright/api/basics.ts:54]: this line breaks SALEOR_137 where after deactivating the user we check API response that user cannot login if account is not active

```
test("TC: SALEOR_137 Admin User should be able to deactivate other user @e2e @staff-members", async () => {
  await staffMembersPage.goToStaffDetailsPage(USERS.userToBeDeactivated.id);
  await staffMembersPage.clickIsActiveCheckbox();
  await staffMembersPage.clickSaveButton();
  await staffMembersPage.basePage.expectSuccessBanner();
  await expect(await staffMembersPage.isActiveCheckbox.isChecked()).toEqual(false);

  const loginViaApiDeactivatedUserResponse = await basicApiService.logInUserViaApi({
    email: USERS.userToBeDeactivated.email,
    password: process.env.E2E_PERMISSIONS_USERS_PASSWORD!,
  });

  await expect(loginViaApiDeactivatedUserResponse.data.tokenCreate.errors[0].code).toEqual(
    "INACTIVE",
  );
});
```

**Review by szczecha — CHANGES_REQUESTED** [review]

File: playwright/api/basics.ts (hunk 1 above)
Comment (yellowee) [reply playwright/api/basics.ts]: fixed https://github.com/saleor/saleor-dashboard/pull/5065/commits/a1af565932df563c4697dfd404e59ce8eae9545e

File: playwright/api/basics.ts [hunk 2]
```diff
@@ -50,7 +50,17 @@ export class BasicApiService {
...
 
-    return loginResponseJson as ApiResponse<TokenCreateResponse>;
+    if (
+      test.info().title ===
+      "TC: SALEOR_137 Admin User should be able to deactivate other user @e2e @staff-members"
+    ) {
+      await expect(errors[0].code).toEqual("INACTIVE");
+    } else {
+      await expect(errors).toEqual([]);
+    }
```
Comment (szczecha) [inline playwright/api/basics.ts:62]: Why do you add assertions here and not in a test that uses the `logInUserViaApi` function?

Comment (yellowee) [reply playwright/api/basics.ts]: You're right - I moved it to auth.setup.ts

**Review by karola312 — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Review by poulch — APPROVED** [review]

**Pull request merged**